.github/workflows/crucible-go-build.yml

```diff
@@ -77,10 +92,16 @@ jobs:
           SOLVER_PKG_VERSION: "snapshot-20210917"
           BUILD_TARGET_OS: ${{ matrix.os }}
 
+      - name: Setup Environment Vars (GHC 8.6.5)
+        if: matrix.ghc == '8.6.5'
+        run: |
+          GHC=haskell.compiler.ghc$(echo ${{ matrix.ghc }} | sed -e s,\\.,,g)
+          echo NS="nix shell github:nixos/nixpkgs/20.09#cabal-install github:nixos/nixpkgs/20.09#$GHC nixpkgs#gmp nixpkgs#zlib nixpkgs#zlib.dev" >> $GITHUB_ENV
       - name: Setup Environment Vars
+        if: matrix.ghc != '8.6.5'
         run: |
           GHC=haskell.compiler.ghc$(echo ${{ matrix.ghc }} | sed -e s,\\.,,g)
-          echo GHC=$GHC >> $GITHUB_ENV
+          echo NS="nix shell github:nixos/nixpkgs/21.05#cabal-install nixpkgs#$GHC nixpkgs#gmp nixpkgs#zlib nixpkgs#zlib.dev" >> $GITHUB_ENV
```

Comment:
Why is `nixpkgs` `21.05` in particular being used here? Intuitively, I would expect the `nixpkgs` version to match what is being installed in the `cachix/install-nix-actions` steps, which would be `20.09` or `unstable`. Again, though, I might be misunderstanding the relationship between these steps.

Reply:
I had to do a bit of experimentation to understand what was happening (thus the different settings you noted in the previous comment), but I believe I've got a handle on it now, so I should be able to do a couple of updates to get everything aligned now.

Basically, the nix tools are evolving from their version 1 forms (`nix-shell`, `nix-build`, `nix-...`) into their version 2 forms (`nix shell`, `nix build`, `nix ...`) which have generally better, but somewhat changed functionality.  In the version 1 form, the installed nixos version would generally dictate which version the packages are being obtained from, so that would have (and used to be) sufficient for determining package sources.  In version 2, the "nixpkgs:" on the command-line causes it to consult the flake registry (`nix registry list`) instead of the `NIX_PATH` set by the installed version (where I was initially assuming the latter).  Thus, the installed version is actually less relevant than the nix shell command-line setting.

It's worth noting (just FYI) that this awkwardness is because we are using a one-line on-the-spot environment generation via `nix shell`, whereas normally the version 2 `nix` commands would look for a flake specification `flake.nix` and a corresponding `flake.lock` file (which form a similar pair to a `.cabal`/`cabal.freeze` file pair) to identify the specific versions of packages desired.

Reply:
OK @RyanGlScott , I'm all done and CI should be good now.

Reply:
Fantastic stuff, and I'm sure I'll come to back to study this later. Thanks again!